### PR TITLE
feat: logs page v2

### PR DESCRIPTION
**Pull request by ogzhanolguncu**
## What does this PR do?

Fixes # (issue)

*If there is not an issue for this, please create one first. This is used to tracking purposes and also helps use understand why this PR exists*

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Test A
- Test B

### Required

- [x] Filled out the "How to test" section in this PR
- [x] Read [Contributing Guide](./CONTRIBUTING.md)
- [x] Self-reviewed my own code
- [x] Commented on my code in hard-to-understand areas
- [x] Ran `pnpm build`
- [x] Ran `pnpm fmt`
- [x] Checked for warnings, there are none
- [x] Removed all `console.logs`
- [x] Merged the latest changes from main onto my branch with `git pull origin main`
- [x] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the Unkey Docs if changes were necessary

## Release Notes

- **New Features**
  - Introduced a `LogsChart` component for visualizing log data with a bar chart.
  - Added a `DatePickerWithRange` component for selecting custom date ranges.
  - Implemented a `ResponseStatus` filter component for managing HTTP response status filters.
  - Launched a `SearchCombobox` for enhanced log search capabilities.
  - Added a `Timeline` component for selecting time ranges for log filtering.
  - Introduced a `LogsFilters` component to streamline log entry filtering.
  - New `LogFooter`, `LogHeader`, `LogMetaSection`, and `LogSection` components for detailed log entry views.
  - Added a `LoadingRow` component to display loading states in the logs table.
  - Introduced a `ResizablePanel` for improved log detail views.
  - Added a `TimestampInfo` component for enhanced timestamp displays.
  - New `Calendar` component for date selection integrated with the UI.
  - Added a `ButtonGroup` component for grouping buttons in the UI.

- **Improvements**
  - Enhanced log querying capabilities with new search parameters and error handling.
  - Updated the `LogsTable` for better performance with large datasets.

- **Bug Fixes**
  - Fixed issues related to log fetching and display, ensuring accurate representation of log data.

- **Documentation**
  - Updated documentation to reflect new components and features introduced in this release.

### CODE REVIEW

**Comment by chronark** [comment]
CleanShot 2024-12-04 at 16 34 15@2x
meta field has weird corners, probably missing an overflow-hidden

File: apps/dashboard/app/(app)/logs/components/logs-table.tsx [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+    );
+  };
+
+  return (
+    <div className="w-full">
+      <div className="grid grid-cols-[166px_72px_20%_1fr] text-sm font-medium text-[#666666]">
+        <div className="p-2 flex items-center">Time</div>
+        <div className="p-2 flex items-center">Status</div>
+        <div className="p-2 flex items-center">Request</div>
+        <div className="p-2 flex items-center">Message</div>
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/logs-table.tsx:48]:
```suggestion
        <div className="p-2 flex items-center">Response Body</div>
```

File: apps/dashboard/app/(app)/logs/components/logs-table.tsx [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      tableRef.current?.getBoundingClientRect().top ?? 0 + window.scrollY - TABLE_BORDER_THICKNESS,
+    );
+  };
+
+  return (
+    <div className="w-full">
+      <div className="grid grid-cols-[166px_72px_20%_1fr] text-sm font-medium text-[#666666]">
+        <div className="p-2 flex items-center">Time</div>
+        <div className="p-2 flex items-center">Status</div>
+        <div className="p-2 flex items-center">Request</div>
```
Comment (chronark) [inline apps/dashboard/app/(app)/logs/components/logs-table.tsx:47]:
```suggestion
        <div className="p-2 flex items-center">Path</div>
```

**Review by chronark — CHANGES_REQUESTED** [review]
CleanShot 2024-12-04 at 16 43 11@2x
We should not parse the body into key-values, it loses the information that this was actually json, and while we always expect json, the user might mess up and send us something invalid.

I'm good with optimistically calling `JSON.stringify(JSON.parse(body), null, 2)` in a try catch to try and format it nicely but falling back to displaying the raw string

**Comment by ogzhanolguncu** [comment]
> CleanShot 2024-12-04 at 16 43 11@2x
> 
> We should not parse the body into key-values, and more importantly we should't uppercase their keys, it loses the information that this was actually json, and while we always expect json, the user might mess up and send us something invalid.
> 
> I'm good with optimistically calling `JSON.stringify(JSON.parse(body), null, 2)` in a try catch to try and format it nicely but falling back to displaying the raw string

I agree, yeah. Let's do this.

**Comment by ogzhanolguncu** [comment]
> CleanShot 2024-12-04 at 16 33 25@2x the separators are sometimes full width and sometimes not

Those are actually padded from left on purpose, but we can change that if we are not happy.

**Comment by chronark** [comment]
> > CleanShot 2024-12-04 at 16 33 25@2x the separators are sometimes full width and sometimes not
> 
> Those are actually padded from left on purpose, but we can change that if we are not happy.

It looks a bit off and unintentional, but maybe that's just me
It's a nitpick really and might change with vitors design anyways

let's keep it as is for now to move quickly

**Review by chronark — APPROVED** [review]

**Pull request merged**